src/waveform/waveform.cpp

```diff
@@ -193,21 +193,21 @@ void Waveform::readByteArray(const QByteArray& data) {
         m_data[i].filtered.high = use_high ? static_cast<unsigned char>(high.value(i)) : 0;
     }
     m_completion = dataSize;
-    m_bDirty = false;
+    m_saveState = SaveState::Saved;
 }
 
 void Waveform::resize(int size) {
     m_dataSize = size;
     m_textureStride = computeTextureStride(size);
     m_data.resize(m_textureStride * m_textureStride);
-    m_bDirty = true;
+    m_saveState = SaveState::SavePending;
```

Comment:
I just swapped the variable, but globally, it doesn't need to be there. This is a private method called from one place only with two paths, one of them setting to saved, and the other that might better have virgin rather than savepending.
Not sure what you mean with "checking savepending before write access". analyzerdao will only save if it the state is savepending.